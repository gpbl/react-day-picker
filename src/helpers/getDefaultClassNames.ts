import { UI, DayFlag, SelectionState } from "../UI.js";
import type { ClassNames } from "../types/index.js";

/**
 * Get the default class names for the UI elements.
 *
 * @group Utilities
 */
export function getDefaultClassNames(): ClassNames {
  const classNames: Partial<Required<ClassNames>> = {};

  for (const key in UI) {
    classNames[UI[key as keyof typeof UI]] =
      `rdp-${UI[key as keyof typeof UI]}`;
  }

  for (const key in DayFlag) {
    classNames[DayFlag[key as keyof typeof DayFlag]] =
      `rdp-${DayFlag[key as keyof typeof DayFlag]}`;
  }

  for (const key in SelectionState) {
    classNames[SelectionState[key as keyof typeof SelectionState]] =
      `rdp-${SelectionState[key as keyof typeof SelectionState]}`;
  }

  return classNames as Required<ClassNames>;
}
