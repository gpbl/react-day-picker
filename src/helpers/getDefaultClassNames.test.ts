import { getDefaultClassNames } from "./getDefaultClassNames";

test("should return the default classnames", () => {
  expect(getDefaultClassNames()).toStrictEqual({
    button_next: "rdp-button_next",
    button_previous: "rdp-button_previous",
    calendar: "rdp-calendar",
    caption_label: "rdp-caption_label",
    chevron: "rdp-chevron",
    day: "rdp-day",
    disabled: "rdp-disabled",
    dropdown_nav: "rdp-dropdown_nav",
    dropdown_root: "rdp-dropdown_root",
    dropdown: "rdp-dropdown",
    focusable: "rdp-focusable",
    focused: "rdp-focused",
    footer: "rdp-footer",
    multiple_months: "rdp-multiple_months",
    week_numbers: "rdp-week_numbers",
    hidden: "rdp-hidden",
    month_caption: "rdp-month_caption",
    month_wrapper: "rdp-month_wrapper",
    month: "rdp-month",
    weeks: "rdp-weeks",
    months_dropdown: "rdp-months_dropdown",
    months: "rdp-months",
    nav: "rdp-nav",
    no_weekdays: "rdp-no_weekdays",
    outside: "rdp-outside",
    range_end: "rdp-range_end",
    range_middle: "rdp-range_middle",
    range_start: "rdp-range_start",
    selected: "rdp-selected",
    today: "rdp-today",
    week_number: "rdp-week_number",
    week: "rdp-week",
    weekday: "rdp-weekday",
    weekdays: "rdp-weekdays",
    years_dropdown: "rdp-years_dropdown"
  });
});
