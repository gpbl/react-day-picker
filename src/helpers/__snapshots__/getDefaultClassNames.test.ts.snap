// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return the default classnames 1`] = `
{
  "button_next": "rdp-button_next",
  "button_previous": "rdp-button_previous",
  "caption": "rdp-caption",
  "caption_dropdowns": "rdp-caption_dropdowns",
  "caption_label": "rdp-caption_label",
  "day": "rdp-day",
  "disabled": "rdp-day_disabled",
  "dropdown": "rdp-dropdown",
  "dropdown_nav": "rdp-dropdown_nav",
  "dropdown_root": "rdp-dropdown_root",
  "excluded": "rdp-day_excluded",
  "focusable": "rdp-day_focusable",
  "focused": "rdp-day_focused",
  "footer": "rdp-footer",
  "multiple_months": "rdp-multiple_months",
  "has_weeknumbers": "rdp-has_weeknumbers",
  "hidden": "rdp-day_hidden",
  "hide_weekdays": "rdp-hide_weekdays",
  "month_caption": "rdp-month_caption",
  "month": "rdp-month",
  "month_wrapper": "rdp-month_wrapper",
  "weeks": "rdp-weeks",
  "months": "rdp-months",
  "months_dropdown": "rdp-months_dropdown",
  "nav": "rdp-nav",
  "outside": "rdp-day_outside",
  "range_end": "rdp-day_range_end",
  "range_middle": "rdp-day_range_middle",
  "range_start": "rdp-day_range_start",
  "rdp": "rdp",
  "select": "rdp-select",
  "selected": "rdp-day_selected",
  "today": "rdp-day_today",
  "week": "rdp-week",
  "weekday_column_header": "rdp-weekday",
  "weekdays": "rdp-weekdays",
  "weeknumber_row_header": "rdp-weeknumber",
  "years_dropdown": "rdp_years_dropdown",
}
`;
